src/ai/helpers/openai/GetRequestIdLogTrafficEventPolicy.cs

```diff
@@ -0,0 +1,54 @@
+using System;
+using System.ClientModel.Primitives;
+using System.IO;
+using System.Text;
+
+namespace Azure.AI.Details.Common.CLI;
+
+public class GetRequestIdLogTrafficEventPolicy : TrafficEventPolicy
+{
+    public GetRequestIdLogTrafficEventPolicy(string filepath, bool outputId, bool addId)
+    {
+        OnRequest += (sender, request) => LogRequest(request);
+        OnResponse += (sender, response) => LogResponse(response, filepath, outputId, addId);
+    }
+
+    private static void LogRequest(PipelineRequest request)
+    {
+    }
+
+    private static void LogResponse(PipelineResponse response, string filepath, bool outputId, bool addId)
+    {
+        string apimRequestId = null;
+        foreach ((string headerName, string headerValue) in response.Headers)
+        {
+            if (headerName.Equals("apim-request-id", StringComparison.OrdinalIgnoreCase))
+            {
+                apimRequestId = headerValue;
+                break;
+            }
+        }
+         if (!string.IsNullOrWhiteSpace(apimRequestId))
+        {
+            // Determine the write mode: append or overwrite
+            FileMode mode = addId ? FileMode.Append : FileMode.Create;
+
+            try
+            {
+                using (FileStream fs = new FileStream(filepath, mode))
```

Comment:
Should use `FileHelpers.AppendAllText` or `FileHelpers.WriteAllText` (they handle other scenarios, things like if you pass in `-` it'll use STDOUT, or, if you pass in `@blah` it'll put it in the `.ai/data` directory that it finds ... )